packages/tools/addon/components/cs-branch-control.js

```diff
@@ -1,29 +1,51 @@
 import { computed } from '@ember/object';
 import { inject as service } from '@ember/service';
 import Component from '@ember/component';
+import { task } from 'ember-concurrency';
 import layout from '../templates/components/cs-branch-control';
 import { defaultBranch } from '@cardstack/plugin-utils/environment';
+import qs from 'qs';
 
 export default Component.extend({
   layout,
+  defaultBranch,
   classNames: ['cs-branch-control'],
-  classNameBindings: ['enabled'],
   tools: service('cardstack-tools'),
+  data: service('cardstack-data'),
 
-  enabled: computed('tools.branch', function() {
-    return !!this.get('tools.branch');
+  'data-test-branch-control': true,
+  'data-test-branch-control-branch-count': computed('branches', function() {
+    return this.get('branches.length');
   }),
 
-  previewing: computed('enabled', 'tools.branch', function() {
-    return this.get('enabled') && defaultBranch !== this.get('tools.branch');
+  willInsertElement() {
+    this._super(...arguments);
+    this.get('fetchBranches').perform();
+  },
+
+  fetchBranches: task(function * () {
+    // checking if the branches() fucntion is available on this version of the data service
+    if((typeof this.get('data').branches) === 'function') {
```

Comment:
Do we actually need this?

Reply:
well, that's a good question 😂 This is my attempt to work around the fact that this package will depend on the features in https://github.com/cardstack/cardstack/pull/535 and won't work until there is a version of the data package **released** and then that is the version that the current app has installed. Essentially depending on a feature like this across packages will cause there to be a **minimum version** of the `cardstack-data` package that is supported. Any other versions would cause a runtime error because of a `undefined is not a function` error.

Maybe I'm being a bit too "safe" here but I couldn't think of a way to get this working in a satisfactory way otherwise 🤔  I would love to get @habdelra and/or @ef4's opinions on this, I'm sure this idea of "minimum supported versions" must have come up in the past at some point.

Reply:
Isn't the way to make this safe, just updating the `tools`' packages [`@cardstack/data` dependency`](https://github.com/cardstack/cardstack/blob/master/packages/tools/package.json#L25) to the version that includes the `branches` method?

Reply:
🤦‍♂️ never mind what I said above - the service is of course coming from the app's dependency as you say…

Reply:
I'm going to merge this PR with https://github.com/cardstack/cardstack/pull/535 because they are going to need to be merged and released at the same time anyway